### DISCUSSION TITLE
Reloaded-II requires "Microsoft.WindowsDesktop.App" version 9.0.0 when install on Linux (Pop OS)

### DISCUSSION THREAD

**Discussion by tdanh2k** [Reloaded-Project/Reloaded-II#490] [Q&A] [answered]
Is Reloaded-II requires "Microsoft.WindowsDesktop.App" on Linux? If so, how can I install it?

**Comment by Sewer56** [comment]
You need to set `DOTNET_ROOT` environment variable to empty. i.e. `DOTNET_ROOT="" `, so Wine doesn't try using the dotnet install from the host OS.

The `Setup-Linux.exe` installer should have automatically done that to your WINEPREFIX, however. Not sure why it isn't kicking in.

**Reply by tdanh2k** [reply]
> You need to set `DOTNET_ROOT` environment variable to empty. i.e. `DOTNET_ROOT="" `, so Wine doesn't try using the dotnet install from the host OS

The thing is, i was following the official Linux Setup, which uses Protontricks and does not specify anything about `DOTNET_ROOT="" `, which confused me. I did try to find The .Net Desktop Runtime for Linux which does not appear to be supported.

**Reply by Sewer56** [reply]
What happens under Wine is.

1. Wine inherits the environment variables of the host system.
2. The .NET Runtime allows you to override its expected location with the `DOTNET_ROOT` environment variable.
3. When you have a native Linux installation of .NET, the package sets `DOTNET_ROOT` as many distros install the runtime in a non-default location.

This is problematic, because when you try to run a .NET application in Wine, it looks at the native Linux one, rather than the Windows one.

The `Setup-Linux.exe` installer sets an [environment variable](https://github.com/Reloaded-Project/Reloaded-II/blob/0037b18e212607c981c085d14aa6bcddc3f26b5b/source/Reloaded.Mod.Installer.Lib/MainWindowViewModel.cs#L115) to explicitly state `DOTNET_ROOT` to be in `Program Files/dotnet` to avoid it inheriting the value from the Linux host. I suspect something is borked around that.

**Reply by tdanh2k** [reply]
Is Protontricks uses Wine under the hood? If not, then is there anyway I can do to mitigate this issue?

**Comment by Sewer56** [comment]
Proton is a modified version of Wine.

Try running the `Exec` command from the `.desktop` file shortcut that's on your Desktop. The `.desktop` file is a text file so you can copy and paste it.

If that doesn't work, let me know what happens if you run the command with `DOTNET_ROOT="" ` before it on the same line in the terminal.

**Reply by tdanh2k** [reply]
![image](https://github.com/user-attachments/assets/64a8a6cf-6130-4275-85a5-71c09faa8d87)
Did I do it right?

**Reply by tdanh2k** [reply]
No, it doesn't work for me

**Reply by Sewer56** [reply]
I didn't get notified for reply with screenshot, oddly.

Anyway I was thinking of copying the original command to your terminal and then prepending `DOTNET_ROOT="" ` before it.

So it would be `DOTNET_ROOT="" protontricks-launch ...`

**Comment by Sewer56** [comment, marked as answer]
Actually, I think I may know the issue.

- https://github.com/dotnet/install-scripts/issues/549

Microsoft servers are serving RC2 instead of the full release when you request `9.0.0`, meaning the installer may have downloaded an outdated runtime.

- [.NET 9 x64 Desktop Runtime](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-9.0.0-windows-x64-installer)
- [.NET 9 x86 Desktop Runtime](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-9.0.0-windows-x86-installer)

Try a manual install through these links, right click, open the protontricks GUI and install into the game's WINEPREFIX.

**Reply by tdanh2k** [reply]
And when it shows the dialog that requires you to download the RC2 version, DO NOT download it. Just close the dialog
